cs/src/core/Index/FASTER/FASTERImpl.cs

```diff
@@ -1439,6 +1509,33 @@ private void BlockAllocate(int recordSize, out long logicalAddress)
             }
         }
 
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private void BlockAllocateReadCache(int recordSize, out long logicalAddress)
+        {
+            logicalAddress = readcache.Allocate(recordSize);
```

Comment:
>readcache.Allocate(recordSize); [](start = 29, length = 31)

I see the allocate calls on the readcache, but I don't see any cleanup. Where do we manage the circular nature of the readcache?

Reply:
You are right, the cleanup functionality is still not added. This is still WIP (work in progress). We will need the cleanup functionality before we can merge.

Reply:
got it

---
In reply to: [259940390](https://github.com/Microsoft/FASTER/pull/92#discussion_r259940390) [](ancestors = 259940390)

Reply:
Cleanup functionality (beta) added to the PR.